src/cplscheme/CouplingData.hpp

```diff
@@ -45,6 +43,12 @@ class CouplingData {
   ///  True, if the data values of this CouplingData require to be initialized by this participant.
   const bool requiresInitialization;
 
+  void initializeStorage();
+
+  void moveToNextWindow();
+
+  void storeDataInWaveform();
+
```

Comment:
Wouldn't it be easier to read if the first two functions also somehow have "waveform" in their name? At least the first one, `initializeWaveformStorage`.

Don't forget the documentation :grin: